index_configuration_tool/tests/test_search_endpoint.py

```diff
@@ -0,0 +1,66 @@
+import copy
+import unittest
+
+import responses
+from responses import matchers
+
+import search_endpoint
+
+# Constants
+TEST_ENDPOINT = "http://test/"
+INDEX1_NAME = "index1"
+INDEX2_NAME = "index2"
+SETTINGS_KEY = "settings"
+INTERNAL_INDEX_KEY = "index"
+TEST_DATA = {
+    INDEX1_NAME: {
+        SETTINGS_KEY: {
+            INTERNAL_INDEX_KEY: {
+                "uuid": "test",
+                "version": 1
+            },
+            "bool_key": True,
+            "str_array": ["abc", "x y z"],
+        },
+        "mappings": {
+            "str_key": "string value",
+            "num_key": -1
+        }
+    },
+    INDEX2_NAME: {
+        SETTINGS_KEY: {},
+        "mappings": {}
+    }
+}
+
+
+class TestSearchEndpoint(unittest.TestCase):
+    def setUp(self) -> None:
+        self.test_data_without_internal = copy.deepcopy(TEST_DATA)
+        # Remove internal data
+        for i in self.test_data_without_internal.keys():
+            self.test_data_without_internal[i][SETTINGS_KEY].pop(INTERNAL_INDEX_KEY, None)
+
+    @responses.activate
+    def test_fetch_all_indices(self):
+        # Set up GET response
+        responses.get(TEST_ENDPOINT + "*", json=TEST_DATA)
+        # Now send request
+        index_data = search_endpoint.fetch_all_indices(TEST_ENDPOINT)
+        self.assertEqual(2, len(index_data.keys()))
+        # Test that internal data has been filtered
+        self.assertTrue(INTERNAL_INDEX_KEY in index_data[INDEX1_NAME][SETTINGS_KEY])
+        self.assertFalse(index_data[INDEX1_NAME][SETTINGS_KEY][INTERNAL_INDEX_KEY])
```

Comment:
I'm not sure I understand this combo of test conditions:
we're testing that the key _is_ present, but the value is gone (or falsey --can we test explicitly for None if that's the expectation?) -- is that correct?  That doesn't match with my understanding of `pop`, it generally deletes the key as well as the value, so--if I'm reading this correctly--I'm surprised it passes.

Reply:
Fixed this test to be clearer. 

The purpose of these assertions is to verify that the `index` key is still present (since this contains other important settings such as `number_of_replicas`) but internal fields have been filtered out. Note that the [filtering logic](https://github.com/kartg/opensearch-migrations/blob/indexConfigurationTool/index_configuration_tool/search_endpoint.py#L23) doesn't `pop` the index key, but the individual internal settings within it.

Since the input data in this revision _only_ had internal keys, the `assertFalse` check was checking for an empty `dict`. I've added a non-internal key now to make this clearer and moved to a more strongly-typed assert.